documents/sd/chapter/Classes/Common/Events/Events.tex

```diff
@@ -11,101 +11,314 @@ \subsection*{MORR.Shared.Events}
+        \end{method}
+    \end{methods}
+
+\end{interface}
+
+\begin{absclass}{SupportDeserializationEventQueue<T>}
+	\clsdiagram[width=0.8\textwidth]{resources/Classes/Common/Shared/Events/Queue/SupportDeserializationEventQueue.png}
+	
+	\clsdcl{public abstract class SupportDeserializationEventQueue<T> : BaseEventQueue<T>, IReadOnlyEventQueue<T>, ISupportDeserializationEventQueue<T> where T : Event}
+	
+	\clsdsp{Provides a single-writer-multiple-reader event queue for event types which may be deserialized for processing.}
+	
+	\begin{methods}
+		\begin{method}{public void Enqueue(object @event)}{Asynchronously enqueues a new event of type T.}
             \begin{parameters}
                 \para{T @event}{The event to enqueue.}
             \end{parameters}
+            \begin{exceptions}
+            	\excp{ArgumentException}{Exception is thrown when event type is not T}
+            \end{exceptions}
         \end{method}
     \end{methods}
 \end{absclass}
+	
+\subsection*{MORR.Shared.Events.Queue.Strategy}
 
+\begin{class}{ChannelConsumingException}
+	\clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/ChannelConsumingException.png}
+	
+	\clsdcl{public class ChannelConsumingException : Exception}	
+	
+	\clsdsp{A simple Exception that encapsulates errors occuring in ChannelStrategies.}
+	
+\end{class}
+	
 \begin{interface}{IEventQueueStorageStrategy}
-    \clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/IEventQueueStorageStrategy.png}
+    \clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/IEventQueueStorageStrategy.png}
 
     \clsdcl{public interface IEventQueueStorageStrategy<T> where T : Event}
 
     \clsdsp{Provides the backing store of an EventQueue with a specific storage strategy.}
 
-    \begin{methods}
-        \begin{method}{IAsyncEnumerable<T> GetEventsInternal()}{Asynchronously gets all events as concrete type T from this storage strategy.}
-            \return{IAsyncEnumerable<T>}{A stream of events of type T.}
-        \end{method}
-        \begin{method}{void Enqueue(T @event)}{Asynchronously enqueues a new event of type T with this storage strategy.}
-            \begin{parameters}
-                \para{T @event}{The event to enqueue.}
-            \end{parameters}
-        \end{method}
-    \end{methods}
 \end{interface}
 
-\begin{class}{RingBufferStorageStrategy}
-    \clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/RingBufferStorageStrategy.png}
+\subsubsection*{MORR.Shared.Event.Queue.Strategy.MultiConsumer}
 
-    \clsdcl{public class RingBufferStorageStrategy<T> : IEventQueueStorageStrategy<T> where T : Event}
+\begin{class}{BoundedMultiConsumerChannelStrategy<TEvent>}
+	\clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/MultiConsumer/BoundedMultiConsumerChannelStrategy.png}
+	
+	\clsdcl{public class BoundedMultiConsumerChannelStrategy<TEvent> : 
```

Comment:
Probably enforce a newline at the end, as currently the text clips out of the document.